### PR TITLE
More tests, fix division by two logic error

### PR DESCRIPTION
**Pull request by tkoolen**
On top of #6.

Add a more rigorous constraint violation test and include some of the older tests. Turns out the division by two from https://github.com/joehuchette/MultilinearOpt.jl/pull/4#discussion_r264458998 *was* wrong. I also verified using this test that quadratic terms (`x == y`) work, so I removed that assertion error. Granted, the relaxation could probably be better in that case, but at least it works. Note also that the `:Unary` method appears to be broken currently.

### CODE REVIEW

File: test/constraint_violation.jl [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        method::Symbol, disc_level::Integer, rng::AbstractRNG, atol::Real, num_tests::Integer)
+    m = Model(with_optimizer(Gurobi.Optimizer, OutputFlag=0))
+    @variable m x lower_bound=xbounds[1] upper_bound=xbounds[2]
+    @variable m y lower_bound=ybounds[1] upper_bound=ybounds[2]
+    @variable m z
+    @constraint m z == f(x, y)
+    relaxbilinear!(m, method=method, disc_level=disc_level)
+    for i = 1 : num_tests
+        x̄ = xbounds[1] + rand(rng) * (xbounds[2] - xbounds[1])
+        ȳ = xbounds[1] + rand(rng) * (xbounds[2] - xbounds[1])
```
Comment (joehuchette) [inline test/constraint_violation.jl]: xbounds --> ybounds

File: test/constraint_violation.jl [hunk 2]
```diff
@@ -0,0 +1,65 @@
+using MultilinearOpt
+using JuMP
+using Gurobi
+using Test
+using Random
+using LinearAlgebra
+
+function test_constraint_violation(f, xbounds::NTuple{2}, ybounds::NTuple{2};
+        method::Symbol, disc_level::Integer, rng::AbstractRNG, atol::Real, num_tests::Integer)
```
Comment (joehuchette) [inline test/constraint_violation.jl]: The implicit assumption here is that that ``rng`` returns values in [0,1], correct?

**Review by joehuchette — CHANGES_REQUESTED** [review]
Looks great, thanks! Only thing to change is what looks like a copy/paste error.

File: test/constraint_violation.jl (hunk 1 above)
Comment (tkoolen) [reply test/constraint_violation.jl]: Oops! Good catch.

File: test/constraint_violation.jl (hunk 2 above)
Comment (tkoolen) [reply test/constraint_violation.jl]: That `rand` does, yes. The documentation of `rand` specifies that the values are in [0, 1).

File: test/constraint_violation.jl (hunk 1 above)
Comment (tkoolen) [reply test/constraint_violation.jl]: Fixed.

**Comment by joehuchette** [comment]
Thanks!

**Pull request merged**